pkg/signup/service/signup_service_test.go

```diff
@@ -78,6 +79,8 @@ func (s *TestSignupServiceSuite) TestSignup() {
 		require.True(s.T(), states.VerificationRequired(&val))
 		require.Equal(s.T(), "a7b1b413c1cbddbcd19a51222ef8e20a", val.Labels[toolchainv1alpha1.UserSignupUserEmailHashLabelKey])
 		require.Empty(s.T(), val.Annotations[toolchainv1alpha1.SkipAutoCreateSpaceAnnotationKey]) // skip auto create space annotation is not set by default
+		require.NotEmpty(s.T(), val.Annotations)
+		require.Equal(s.T(), requestTime.Format(time.RFC3339), val.Annotations[toolchainv1alpha1.UserSignupRequestReceivedTimeAnnotationKey])
```

Comment:
That assumes that the test is finished within 1s. Which should be the case in a normal test run. But at least maybe worth adding a comment? Just in case, for whatever reason, the test fails due to some delay (during test debugging for example), so it could help to understand why it failed.

Reply:
or should we just assert that the annotation is there and has a valid `time.RFC3339` formatted value ?

Reply:
Not really, this test doesn't rely on the execution time anyhow - the requestTime is set once at the beginning of the suite

https://github.com/codeready-toolchain/registration-service/blob/b9ec4a9145ad739cd8befffdd2988e8f7b670cea/pkg/signup/service/signup_service_test.go#L68

and the same value is set in the context

https://github.com/codeready-toolchain/registration-service/blob/b9ec4a9145ad739cd8befffdd2988e8f7b670cea/pkg/signup/service/signup_service_test.go#L110

and the same value is also used for comparison

https://github.com/codeready-toolchain/registration-service/blob/b9ec4a9145ad739cd8befffdd2988e8f7b670cea/pkg/signup/service/signup_service_test.go#L83

So, it doesn't matter how long the test runs, it will always pass. (I've tested it locally)